src/js/main.js

```diff
@@ -2,27 +2,57 @@ import spreadsheet from './spreadsheet';
 import userColsAndRows from './helpers/userColsAndRows';
 import toggleDarkMode from './darkModeToggle/toggleDarkMode.mjs';
 import { addCellTargetingEvents } from './spreadsheet/cellNavigation';
-import { initDB } from './spreadsheet/db.js';
+import { initDB, saveCellValue, getCellValue } from './spreadsheet/db.js';
 
 const spreadsheetContainer = document.querySelector('#spreadsheetContainer');
 
-// indexedDB
+// Function to load initial data from IndexedDB
+async function loadInitialData(cols, rows) {
+  const data = [];
+  for (let row = 0; row < rows; row++) {
+    const rowData = [];
+    for (let col = 0; col < cols; col++) {
+      const cellId = `cell-${row}-${col}`;
+      const cellValue = await getCellValue(cellId);
+      rowData.push(cellValue !== null ? cellValue : '');
+    }
+    data.push(rowData);
+  }
+  return data;
+}
+
+// Initialize the spreadsheet with data from IndexedDB or default data
 initDB()
-  .then(() => {
+  .then(async () => {
     console.log('IndexedDB initialized');
 
     // DarkMode
     toggleDarkMode();
 
     const [cols, rows] = userColsAndRows();
 
+    // Load initial data
+    const initialData = await loadInitialData(cols, rows);
+
     // Create and append the spreadsheet to the container
-    spreadsheetContainer.append(spreadsheet(cols, rows));
+    const hot = spreadsheet(cols, rows, initialData);
+    spreadsheetContainer.append(hot);
 
     addCellTargetingEvents('#spreadsheetContainer table', (col, row, value) => {
```

Comment:
now a callback is deleted, did you on purpose?

Reply:
It is not deleted. It is refactored and stored in a constant to make it reusable if needed elsewhere